pkg/spoke/hubclientcert/controller.go

```diff
@@ -103,34 +103,30 @@ func NewClientCertForHubController(
 	}
 
 	return factory.New().
-		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
-			accessor, err := meta.Accessor(obj)
-			if err != nil {
-				return irrelevantSecretKey
-			}
-			// we want to only care about the hub kubeconfig secret, others are irrelevant
-			if accessor.GetNamespace() == hubKubeconfigSecretNamespace && accessor.GetName() == kubeconfigSecretName {
+		WithFilteredEventsInformersQueueKeyFunc(
+			func(obj runtime.Object) string {
+				accessor, _ := meta.Accessor(obj)
 				return accessor.GetName()
-			}
-			return irrelevantSecretKey
-		}, spokeSecretInformer.Informer()).
+			},
+			func(obj interface{}) bool {
```

Comment:
I'm open to pushing this into a function in library-go for, `filterByInstance(ns, name) func` or some such.